challenge/contact.py

```diff
@@ -1,188 +0,0 @@
-#!/usr/bin/env python
-#
-from selenium import webdriver
-from urllib.parse import urlparse,urlunparse
-import urllib.request
-
-import logging
-import pdb
-import re
-import sys
-import platform
-
-class ui:
-    """ web access with ui capabilities """
-    def __init__(self,hack=''):
-        """ Selenium web driver can be set up 'headless' browser on some remote system, which could improve performance or allow running multiple copies"""
-        if hack:
-            self.web = webdriver.Firefox(webdriver.FirefoxProfile(hack))
-        else:
-            self.web = webdriver.Firefox()
-
-        self.actions = webdriver.common.action_chains.ActionChains(self.web)
-        #self.scan.actions.getCurrentWindow().focus()
-        for _ in range(12):
-            self.actions.send_keys(webdriver.common.keys.Keys.TAB)
-        #for _ in range(3)
-        #    self.actions.send_keys(webdriver.common.keys.Keys.PAGE_DOWN)
-
-        # perform actions later
-    def __enter__(self):
-        return self
-    def __exit__(self,t,v,s):
-        self.web.quit()
-
-scheduled = set() # deque? avoid duplicate scheduling
-done = set() # web pages already examined
-emails = set() # email addresses found
-
-def normalized(link):
```

Comment:
If you are referring to class "ui", I want to use a "with" statement to make sure the browser gets closed, even on "abend" (a beautiful German word).

If you mean "normalized", I wanted to avoid scanning the same pages with different query params or URL fragment specifiers.